microprofile/graphql/server/src/main/java/io/helidon/microprofile/graphql/server/SchemaDirective.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2019, 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.graphql.server;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+/**
+ * The representation of a GraphQL directive.
+ */
+class SchemaDirective implements ElementGenerator {
+
+    /**
+     * The name of the directive.
+     */
+    private final String name;
+
+    /**
+     * The list of arguments for the directive.
+     */
+    private final List<SchemaArgument> listSchemaArguments;
+
+    /**
+     * The locations the directive applies to.
+     */
+    private final Set<String> setLocations;
+
+    /**
+     * Construct a {@link SchemaDirective}.
+     *
+     * @param builder the {@link Builder} to construct from
+     */
+    private SchemaDirective(Builder builder) {
+        this.name = builder.name;
+        this.listSchemaArguments = builder.listSchemaArguments;
+        this.setLocations = builder.setLocations;
+    }
+
+    /**
+     * Fluent API builder to create {@link SchemaDirective}.
+     *
+     * @return new builder instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Return the GraphQL schema representation of the element.
+     *
+     * @return the GraphQL schema representation of the element.
+     */
+    @Override
+    public String getSchemaAsString() {
+        StringBuilder sb = new StringBuilder("directive @" + name());
+
+        if (listSchemaArguments.size() > 0) {
+            sb.append("(");
+            AtomicBoolean isFirst = new AtomicBoolean(true);
+            listSchemaArguments.forEach(a -> {
+                String delim = isFirst.get() ? "" : ", ";
+                isFirst.set(false);
```

Comment:
`String delim = isFirst.getAndSet(false) ? "" : ", ";` But do we need AtomicBoolean at all?

Reply:
Isfirst needs to be (effectivly) final as referenced in lambda, so cannot be boolean or Boolean.